File: detect_cycle.py
def DFS(s):
    visited = [False]*(n)
    visited[s]=True
    stack = []
    S = []
    stack.append(s) 
    S.append(s)
    while stack:
        x = stack.pop(len(stack)-1)
        print(x)
        for v in range(n):
            if L[x][v]:
                if v in S:              #If visiting vertex is already in S , cycle present...
                    return True
                elif not visited[v]:
                    visited[v] = True
                    stack.append(v) 
                    S.append(s)
    return False

L = []
n = int(input("Enter n value : "))
print("Enter a adj_matrix :")
for i in range(n):
    L.append([])
    for j in range(n):
        L[i].append(int(input()))

s = int(input("Enter starting vertex : ")) 
if DFS(s):
    print("Present")
else:
    print("Not present")

File: java/Detect_cycle.java
import java.util.*;

class Detect_cycle
{
    public static boolean DFS(int[][] G,int n,int s)
    {
        int[] visited = new int[n];
        visited[s] = 1;
        ArrayList<Integer> stack = new ArrayList<Integer>();
        ArrayList<Integer> S = new ArrayList<Integer>();
        stack.add(s);
        S.add(s);
        while(!stack.isEmpty())
        {
            int x = stack.remove(stack.size()-1);
            for(int i=0;i<n;i++)
            {
                if(G[s][i]==1 && S.contains(i)) return true;
                else if(G[s][i]==1 && visited[i]==0)
                {
                    visited[i] = 1;
                    stack.add(i);
                    S.add(i);
                }
            }
        }
        return false;
    }

    public static void main(String[] args)
    {
        Scanner jin = new Scanner(System.in);
        int n = jin.nextInt();
        int[][] G = new int[n][n];
        for(int i=0;i<n;i++)
            for(int j=0;j<n;j++)
                G[i][j] = jin.nextInt();
        System.out.println("Enter starting...");
        int s = jin.nextInt();
        System.out.println(DFS(G,n,s)==true?"Present":"NO");
        jin.close();
    }
}